src/renderers/dom/client/ReactMount.js

```diff
@@ -880,9 +880,25 @@ var ReactMount = {
           checksum
         );
 
-        var diffIndex = firstDifferenceIndex(markup, rootMarkup);
+        // because rootMarkup is retrieved from the DOM, various normalizations
+        // will have occurred which will not be present in `markup`. Here,
+        // insert markup into a <div> or <iframe> depending on the container
+        // type to perform the same normalizations before comparing.
+        var normalizer, normalizedMarkup;
+        if (container.nodeType === ELEMENT_NODE_TYPE) {
+          normalizer = document.createElement('div');
+          normalizer.innerHTML = markup;
+          normalizedMarkup = normalizer.innerHTML;
+        } else {
+          normalizer = document.createElement('iframe');
+          document.body.appendChild(normalizer);
```

Comment:
I'd prefer not to, but unfortunately iframes only get a valid contentDocument when in the DOM. I've addressed this by immediately removing it from the DOM.